### PR TITLE
Add slow retrieval attack protection to ECUs

### LINKED ISSUE #190: Slow retrieval attack protection
In the threat model, we state that Uptane protects against slow retrieval attacks. However, full protection against slow retrieval attacks requires cooperation from the network stack which is not described in the standard and probably out of scope of Uptane. For this reason, we recently removed slow retrieval attacks from the TUF specification (see theupdateframework/specification#111). To resolve this, we can do one of two things:

* Remove slow retrieval attacks from the threat model (and perhaps move it to the Deployment Best Practices)

* Add a requirement for ECUs to detect and prevent slow retrieval attacks.

This might be something to discuss for 2.0.

### PR DESCRIPTION
**Pull request by mnm678**
Fixes #190.

I added a description of slow retrieval attack prevention to the in-vehicle requirements.

### CODE REVIEW

**Review by jhdalek55 — APPROVED** [review]
This looks fine to me. @trishankkarthik, @patrickvacek do we need to say anything more?

**Review by patrickvacek — APPROVED** [review]
Looks fine to me. Not sure what else would be worth saying here.

**Comment by iramcdonald** [comment]
Hi Marina,

Thanks.  This text looks good to me.

Cheers,
- Ira

*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Anchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: blueroofmusic@gmail.com
<blueroofmusic@gmail.com>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

File: uptane-standard.md
Comment (trishankatdatadog) [inline uptane-standard.md:648]: Provide a warning to whom? To other ECUs or itself?

**Comment by iramcdonald** [comment]
Hi Trishank,

Good catch.

"Provide a warning" clearly drags in secure logs and secure notifications
(both of which
are underspecified in Uptane, I think).

If the "warning" doesn't go further than the Primary (i.e., is never
reported to the Director
or other back-end infrastructure systems), the "warning" is useless for
cybersecurity.

Cheers,
- Ira

*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Anchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: blueroofmusic@gmail.com
<blueroofmusic@gmail.com>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

File: uptane-standard.md
Comment (mnm678) [reply uptane-standard.md]: I updated the text to ensure the warning is sent to the director so that it can be handled off-vehicle. I'm not sure if it should also be sent to the repo where the download originates (possibly the image repo).

**Comment by iramcdonald** [comment]
Hi Marina,

My two cents.

An Uptane Primary (or relay of a Secondary report) should often *fetch*
software and metadata
from one or more Image Repositories, based on an update campaign from a
Director.  But they
should *never* directly send notifications to Image Repositories.  They
should always send their
notifications to the Director, who should orchestrate
verification/validation of the notifications and
forward them to other systems in the back-end infrastructure as appropriate.

Especially because, in real deployments, the Image Repositories visible to
an Uptane Primary
may indeed be servers in a third-party Content Delivery Network (CDN) or a
commercial Cloud
service provider (much less secure!!!) for scalability of software
distribution.

Cheers,
- Ira

*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Anchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: blueroofmusic@gmail.com
<blueroofmusic@gmail.com>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

**Comment by trishankatdatadog** [comment]
Thanks, Marina and Ira! One more thing: should we clarify that this alert should be sent as part of the ECU version reports (I forget what they are called now)?

**Comment by mnm678** [comment]
> One more thing: should we clarify that this alert should be sent as part of the ECU version reports (I forget what they are called now)?

The version report would be one place for this alert, but there are other valid ways to send it, so I hesitate to specify how the alert is sent.

**Comment by trishankatdatadog** [comment]
> The version report would be one place for this alert, but there are other valid ways to send it, so I hesitate to specify how the alert is sent.

Got it. Maybe we should still suggest it anyway, rather than suggest that this "needs" a different mechanism? I will leave this decision to Patrick and Ira.

**Comment by iramcdonald** [comment]
Hi,

We should explicitly say that the notification is always sent to the
Director (descriptive
behavior, rather than a new shall requirement) and then say, for example in
an ECU
version report.

Cheers,
- Ira

*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Anchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: blueroofmusic@gmail.com
<blueroofmusic@gmail.com>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

**Comment by patrickvacek** [comment]
> We should explicitly say that the notification is always sent to the Director (descriptive behavior, rather than a new shall requirement) and then say, for example in an ECU version report.

Makes sense to me. Should we add the SHOULD/SHALL in a future version? And can the option of using the vehicle version manifest be a MAY, or is it best to leave it as purely just as an example?

**Comment by iramcdonald** [comment]
Hi Patrick,

My two cents - in v2.0 we should have a SHALL for the detection and
reporting and a SHOULD for use of ECU version info.

For now,  we should *not* add a MAY for the use of ECU version info
(because we really mean SHOULD - this is the most reliable way to
report the issue).

Cheers,
- Ira

*Ira McDonald (Musician / Software Architect)*

*Chair - SAE Trust Antsnchors and Authentication TF*
*Co-Chair - TCG Trusted Mobility Solutions WG*

*Co-Chair - TCG Metadata Access Protocol SG*








*Chair - Linux Foundation Open Printing WGSecretary - IEEE-ISTO Printer
Working GroupCo-Chair - IEEE-ISTO PWG Internet Printing Protocol WGIETF
Designated Expert - IPP & Printer MIBBlue Roof Music / High North
Inchttp://sites.google.com/site/blueroofmusic
<http://sites.google.com/site/blueroofmusic>http://sites.google.com/site/highnorthinc
<http://sites.google.com/site/highnorthinc>mailto: blueroofmusic@gmail.com
<blueroofmusic@gmail.com>(permanent) PO Box 221  Grand Marais, MI 49839
906-494-2434*

**Review by trishankatdatadog — APPROVED** [review]
LGTM, thanks, Marina, Ira, and Patrick!

**Comment by jhdalek55** [comment]
This needs one more approval before merging. @patrickvacek, can you approve and then can someone merge?

**Review by patrickvacek — APPROVED** [review]

**Pull request merged**
